azext_iot/_factory.py

```diff
@@ -75,9 +76,13 @@ def __init__(self, target, device_id=None, auth_override=None):
         self.auth_override = auth_override
 
         # This initialization will likely need to change to support more variation of SDK
-        self.sas_uri = self.target["entity"]
+        self.sas_uri = self.target.get("entity", IOTDPS_PROVISIONING_HOST)
         self.endpoint = "https://{}".format(self.sas_uri)
-        if self.device_id:  # IoT Hub base endpoint stays the same
+
+        # Base endpoints stay the same
+        if self.device_id and self.target.get("idscope"):
+            self.sas_uri = "{}/registrations/{}".format(self.target["idscope"], self.device_id)
+        elif self.device_id:
```

Comment:
nit, `self.device_id` is already evaluated in the first if, this if/elif can be improved to not have to evaluate the same condition twice.

Reply:
removed